src/Commands/TerminusCommand.php

```diff
@@ -7,32 +7,31 @@
 use Psr\Log\LoggerAwareTrait;
 use Psr\Log\LoggerInterface;
 use Robo\Contract\IOAwareInterface;
+use Robo\Contract\ConfigAwareInterface;
+use Robo\Common\ConfigAwareTrait;
 use Robo\Common\IO;
 use Terminus\Models\Auth;
 
-abstract class TerminusCommand implements IOAwareInterface, LoggerAwareInterface
+abstract class TerminusCommand implements IOAwareInterface, LoggerAwareInterface, ConfigAwareInterface
 {
     use LoggerAwareTrait;
+    use ConfigAwareTrait;
     use IO;
 
     /**
      * @var boolean True if the command requires the user to be logged in
      */
     protected $authorized = false;
-    /**
-     * @var Config
-     */
-    protected $config;
 
     /**
      * TerminusCommand constructor
      */
     public function __construct()
     {
-        $this->config = new Config();
-        if ($this->authorized) {
-            $this->ensureLogin();
-        }
+        // TODO: Cannot log in until our dependencies are inflected
```

Comment:
Remember, setter injection doesn't happen until after the constructor is called, so if ensureLogin() depends on any injected dependencies, then it cannot be called during the constructor.

Also, all Terminus commands are constructed at the beginning of the program, before Symfony Console has decided which command is running.  If you call ensureLogin during the constructor, then it will be called many times, once for each command that requires a login, regardless of whether or not the selected command needs to log in.

Maybe we could add an @authorized annotation, and have a hook call ensureLogin if the selected command needs to log in?

Is ensureLogin machine token only now, or can you still log in with username and password?  Is the password supplied via an option?  If so, then ensureLogin would need to be called after interact, to allow the user the chance to enter it if it was not provided via the commandline option.

Not sure how things are going to be working in 1.x, though.